packages/app-mobile/components/NoteBodyViewer/hooks/useOnMessage.ts

```diff
@@ -22,6 +21,10 @@ export default function useOnMessage(onCheckboxChange: Function, noteBody: strin
 		} else if (msg.startsWith('error:')) {
 			console.error(`Webview injected script error: ${msg}`);
 		} else {
+			// msg is broken at this point if it contained special characters
+			// see: https://github.com/laurent22/joplin/issues/4494
+
+			msg = event.nativeEvent.data;
 			onJoplinLinkClick(msg);
```

Comment:
Ok now that I see the changes again I understand why the test couldn't demonstrate much, since you're just passing the message directly the click handler, without any unescaping.

But I feel we aren't getting to the bottom to of the issue. The message is unescaped twice at the top, and from the look of it this unescaping is only relevant when opening a link, but now we've disabled that. So there's lot of unanswered questions: why we're unescaping twice? Maybe it was to go around a bug in the webview, which no longer exists? Maybe it's needed for Android but not iOS (in which case your fix would break in Android)? Basically, I feel we're fixing something but without knowing why we're doing it.

So perhaps you could check this? Do we need to keep the double unescaping at all? What happens if we remove it?

Reply:
One thing I noticed is that this bug exists both on iOS and Android, and this patch shall fix it on both platforms. 

My guess is that instead of double escape we should use `decodeURI(encodeURI(event.nativeEvent.data))` instead. 

I was trying to figure out why you double escaped at the first place, however, I'm having some hard time finding the usage of `useOnMessage` other than when clicking on the link. For example, the if statements in the function seems to suggest that the function also called when `checkboxclick` (I'm assuming the todo checkbox). But when check the checkbox it's not called. 

Could you perhaps let me know when it should be called so I can test out if it will be broken if we change double escape to something else?

Reply:
It is called whenever you click on an external URL from the note viewer. It might well be that we should use decode/encodeURI so yes maybe check if it makes a difference.